### PR TITLE
feat(@angular/cli): drop support for Node.js 12

### PR DESCRIPTION
**Pull request by alan-agius4**
Node.js v12 will become EOL on 2022-04-30. As a result, Angular CLI v14 will no longer support Node.js v12.

BREAKING CHANGE:

Support for Node.js v12 has been removed as it will become EOL on 2022-04-30. Please use Node.js v14.15 or later.

### CODE REVIEW

File: packages/angular_devkit/core/node/host.ts [hunk 1]
```diff
@@ -91,24 +91,7 @@ export class NodeJsAsyncHost implements virtualFs.Host<Stats> {
     return this.isDirectory(path).pipe(
       mergeMap(async (isDirectory) => {
         if (isDirectory) {
```
Comment (clydin) [inline packages/angular_devkit/core/node/host.ts:93]: With the use of only `rm`, the `isDirectory` check should also be able to be removed.

File: .circleci/config.yml [hunk 2]
```diff
@@ -339,7 +339,6 @@ workflows:
             - setup
       - e2e-cli:
           name: e2e-cli
-          nodeversion: '14.15'
```
Comment (clydin) [inline .circleci/config.yml:342]: With the package engine field using 14.15 as the minimum, I think we should consider keeping this node version for the E2E tests.

**Review by clydin — APPROVED** [review]

**Review by dgp1130 — APPROVED** [review]
Thanks for cleaning this up!

File: packages/angular_devkit/core/node/host.ts (hunk 1 above)
Comment (alan-agius4) [reply packages/angular_devkit/core/node/host.ts]: Done

**Pull request merged**